qt/applications/workbench/workbench/app/start.py

```diff
@@ -49,13 +49,14 @@ def qapplication():
 
         argv = sys.argv[:]
         argv[0] = APPNAME  # replace application name
-        # Workaround a segfault with the IPython console when using Python 3.5 + PyQt 5
-        # Without this using this fix the above combination causes a segfault when the IPython
-        # console is started
+        # Workaround a segfault importing readline with PyQt5
+        # This is because PyQt5 messes up pystate (internal) modules_by_index
+        # so PyState_FindModule will return null instead of the module address.
+        # Readline (so far) is the only module that falls over during init as it blindly uses FindModules result
         # The workaround mentioned in https://groups.google.com/forum/#!topic/leo-editor/ghiIN7irzY0
-        # is to ensure readline is imported before the QApplication object is created
-        if sys.version_info[0] == 3 and sys.version_info[1] == 5:
+        if sys.platform == "linux" or sys.platform == "linux2" or sys.platform == "darwin":
```

Comment:
If we started shipping `readline` on Windows then presumably this would then break and have to be tweaked?

Would we be better just wrapping the import in `try ... except ModuleNotFoundError:` so that if it exists it will be imported regardless of the OS?

Reply:
Its not failing with a python exception, IIRC it relies on `readline` being installed externally (say from a .deb) to find it at runtime or fails with a segmentation fault

Reply:
I think I misunderstood. Are we not wanting to import it here, if it is available, before `QApplication` is created so we can can avoid the later segfault?
My meaning was that instead of checking platforms and having to update the list if we ship it on Windows that we use the `ModuleNotFoundException` to indicate if it's not there? Presumably the import works at this point or the current fix wouldn't?

Reply:
Sorry there are two segmentation faults at play:

1. (Windows) If you don't have readline installed (which ships with Python in distros) you will segmentation fault from what I can tell: https://stackoverflow.com/a/51427421
2. Mac + Linux - Trying to import readline after PyQt5 will cause a segmentation fault

Reply:
Ah I see. `readline` actually doesn't exist on Windows rather than us not shipping it. I'm caught up! Annoyingly the documentation doesn't state this like it does in other modules.
Thanks for the clarification.

Reply:
I've suggested if we run into this with CaChannel that we upstream them to use `gnureadline` as at least it comes with statically linked executables from pip - but we can cross that bridge as and when